src/openloco/things/CreateVehicle.cpp

```diff
@@ -0,0 +1,1000 @@
+        base->base_type = thing_base_type::vehicle;
+        base->type = T::VehicleThingType;
+        return reinterpret_cast<T*>(base);
+    }
+
+    // 0x004BA873
+    // esi : vehBogie
+    static void sub_4BA873(openloco::vehicle_bogie* const vehBogie)
+    {
+        vehBogie->var_68 = 0xFFFF;
+        if (vehBogie->reliability != 0)
+        {
+            int32_t reliabilityFactor = vehBogie->reliability / 256;
+            reliabilityFactor *= reliabilityFactor;
+            reliabilityFactor /= 16;
+
+            auto& prng = gprng();
+            int32_t randVal = (prng.rand_next(65535) * reliabilityFactor / 2) / 65536;
+            reliabilityFactor -= reliabilityFactor / 4;
+            reliabilityFactor += randVal;
+            vehBogie->var_68 = static_cast<uint16_t>(std::max(4, reliabilityFactor));
+        }
+    }
+
+    // 0x4AE8F1
+    static openloco::vehicle_bogie* createBogie(const thing_id_t head, const uint16_t vehicleTypeId, const vehicle_object& vehObject, const uint8_t bodyNumber, openloco::vehicle* const lastVeh, const ColourScheme colourScheme)
+    {
+        auto newBogie = createVehicleThing<vehicle_bogie>();
+        newBogie->owner = _updating_company_id;
+        newBogie->head = head;
+        newBogie->body_index = bodyNumber;
+        newBogie->track_type = lastVeh->track_type;
+        newBogie->mode = lastVeh->mode;
+        newBogie->tile_x = -1;
+        newBogie->tile_y = 0;
+        newBogie->tile_base_z = 0;
+        newBogie->var_2E = 0;
+        newBogie->var_2C = 0;
+        newBogie->var_36 = lastVeh->var_36;
+        newBogie->object_id = vehicleTypeId;
+
+        auto& prng = gprng();
+        newBogie->var_44 = prng.rand_next();
+        newBogie->creation_day = current_day();
+        newBogie->var_46 = 0;
+        newBogie->var_47 = 0;
+        newBogie->accepted_cargo_types = 0;
+        newBogie->cargo_type = 0xFF;
+        newBogie->var_51 = 0;
+        newBogie->var_5E = 0;
+        newBogie->var_5F = 0;
+        newBogie->var_60 = 0; // different
+        newBogie->var_61 = 0; // different
+
+        newBogie->var_14 = 1;
+        newBogie->var_09 = 1;
+        newBogie->var_15 = 1;
+
+        newBogie->colour_scheme = colourScheme;
+        lastVeh->next_car_id = newBogie->id;
+        return newBogie;
+    }
+
+    // 0x4AE8F1
+    static openloco::vehicle_bogie* createFirstBogie(const thing_id_t head, const uint16_t vehicleTypeId, const vehicle_object& vehObject, const uint8_t bodyNumber, openloco::vehicle* const lastVeh, const ColourScheme colourScheme)
+    {
+        auto newBogie = createBogie(head, vehicleTypeId, vehObject, bodyNumber, lastVeh, colourScheme);
+        if (newBogie == nullptr) // Can never happen
+        {
+            return nullptr;
+        }
+        newBogie->var_38 = 0;
+
+        int32_t reliability = vehObject.reliability * 256;
+        if (current_year() + 2 > vehObject.designed)
+        {
+            reliability -= reliability / 8;
```

Comment:
add comment. // Remove an eighth of reliability after 2 years past designed life